### PR TITLE
Grade multiple submissions at once

### PR DESCRIPTION
**Pull request by sahi1l**
In the Grading panel, this lets you select multiple students or multiple questions, and shows the answers for all of them at once. This will let instructors compare multiple answers at once, going back and forth between different students' code without constantly hitting the database (a tedious process).

There are undoubtedly subtle problems with this I haven't thought of, but pleeeeease can we have this for the Fall?

### CODE REVIEW

**Comment by bnmnetp** [comment]
I'll do a thorough review when I return.  But my initial impressions are that you did some nice refactoring of some really nasty old code. 💯

**Comment by bnmnetp** [comment]
P.S.   -- I don't know if your work fixes this (longstanding bug), but since you are doing major surgery on this code I'll mention it.

We should be able to select a chapter, then an exercise in that chapter, then (one or more) student so that you can easily check work that is not part of an assignment.  Double bonus points for having this working when I'm back on July 20.

**Comment by sahi1l** [comment]
If I knew how to access a list of all the questions in a given chapter, I could do it I think.

**Comment by bnmnetp** [comment]
The questions table in the database has both chapter and sub chapter information.  The name column of the questions table corresponds to the unique I’d of the question.

**Comment by sahi1l** [comment]
I don't know how to access that in Javascript, though.  There is a variable called `assignmentinfo` defined somehow, but I don't know how to create a corresponding `chapterinfo` variable.  (I still have huge gaps in my understanding.)

**Comment by sahi1l** [comment]
I figured out the chapters problem; it's been added to this pull request.

File: views/admin/grading.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
+{{ response.title="Grading" }}
```
Comment (bnmnetp) [inline views/admin/grading.html]: Should this be removed in favor of the PR I just accepted that sets this in the controller?

File: static/js/admin.js [hunk 2]
```diff
@@ -302,7 +307,7 @@ function getRightSideGradingDiv(element, acid, studentId) {
             var file_div_template = '<pre id="file_div_template" style = "display:none;">template text</pre>;'
             var index;
             for (index = 0; index < data.file_includes.length; index += 1) {
-                if (jQuery('#' + data.file_includes[index].acid).length == 0) {
+                if (jQuery('#' + data.file_includes[index].acid,rightDiv).length == 0) {
```
Comment (bnmnetp) [inline static/js/admin.js]: Is this correct?  What does the ,rightDiv part do?

File: static/js/admin.js [hunk 3]
```diff
@@ -1,65 +1,71 @@
...
-    var select3 = document.getElementById("gradingoption3");
-    var colType = select3.options[select3.selectedIndex].value;
-
-    var col1 = document.getElementById("gradingoption1");
-    var col1val = col1.options[col1.selectedIndex].value;
-
-    var col2 = document.getElementById("gradingoption2");
-    var col2val = col2.options[col2.selectedIndex].value;
+    //This function figures out the parameters to feed to getRightSideGradingDiv, which does most of the work
+    var SEL1 = document.getElementById("gradingoption1");
```
Comment (bnmnetp) [inline static/js/admin.js]: Is there significance to the all caps variable names?

File: views/admin/grading.html (hunk 1 above)
Comment (sahi1l) [reply views/admin/grading.html]: Yeah, I don't know how that snuck in there.

File: static/js/admin.js (hunk 2 above)
Comment (sahi1l) [reply static/js/admin.js]: ,rightDiv requires that the element be a child of rightDiv; there are now multiple rightDivs, one for each problem to grade, and this is easier than giving all the children of each rightDiv unique names.

File: static/js/admin.js (hunk 3 above)
Comment (sahi1l) [reply static/js/admin.js]: Only to make sure I didn't confuse them with the old variables. If this goes against style I can change them to lowercase.

Comment (bnmnetp) [reply static/js/admin.js]: Makes sense, and although I haven't enforced too much in the way of style, it should be easy enough to make them lowercase now.

I plan to set out some style guidelines and variables should definitely be lowercase.

Thanks!

File: static/js/admin.js (hunk 2 above)
Comment (bnmnetp) [reply static/js/admin.js]: gotcha, I've not used that syntax with jQuery before.

**Comment by bnmnetp** [comment]
This should be ready to accept once you get the variable names lowercased!

File: views/admin/grading.html (hunk 1 above)
Comment (sahi1l) [reply views/admin/grading.html]: Removed from the latest commit.

File: static/js/admin.js (hunk 3 above)
Comment (sahi1l) [reply static/js/admin.js]: The latest commit fixes this.

**Comment by sahi1l** [comment]
This should be all set then! Woohoo!

File: static/js/admin.js [hunk 4]
```diff
@@ -409,108 +413,113 @@ function getRightSideGradingDiv(element, acid, studentId) {
...
-
+    else if (val1 == 'chapter' && val2 == 'question') {
+        //FIX: This is where we should get a list of all questions from the chapter
+        //chapters[label] should store a list of all question names
+        //populateQuestions should be a model for this.
+        populateQuestions(col2,chapters[col1val]);
     }
 
-    else if (val == 'student') {
+    else if (VAL1 == 'student') {
```
Comment (bnmnetp) [inline static/js/admin.js]: You missed a few -- that caused grading to break.  But I took care of them.

**Comment by bnmnetp** [comment]
I merged this from the CL.  Thanks @sahi1l

**Comment by sahi1l** [comment]
This will be so helpful next semester, I hope. Thanks for not making me wait for it. :)

What's the CL?

**Comment by bnmnetp** [comment]
Sure.  I'm happy to add these things that will make life better for you and others.

Command Line

**Comment by sahi1l** [comment]
Oh duh of course, I thought it was some Git jargon I didn't know. :)

**Pull request closed without merging**